models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql

```diff
@@ -0,0 +1,159 @@
+        , l.index AS evt_index
+        , l.contract_address AS project_contract_address
+        , NULL AS transfer_id
+        , 1 AS source_chain_id
+        , 10 AS destination_chain_id
+        ,DENSE_RANK() OVER (PARTITION BY tx_hash ORDER BY index ASC) AS tf_index
+        
+        FROM {{ source ('optimism', 'logs') }} l
+        WHERE topic1 = '0xb0444523268717a02698be47d0803aa7468c00acbed2f8bd93a0459cde61dd89' --Deposit Finalized
+        -- Other potentially helpful filters
+        AND (topic2 IS NOT NULL) AND (topic3 IS NOT NULL) AND (topic4 IS NOT NULL) AND (data IS NOT NULL)
+        {% if is_incremental() %}
+        AND block_time > NOW() - interval '14 days'
+        {% endif %}
+        
+        UNION ALL
+        
+        SELECT 
+        'Withdrawal' AS tf_type
+        , l.block_time
+        , l.block_number
+        , l.tx_hash
+        , '0x' || substring(topic3,27,40) AS bridged_token_address
+        , conv(
+            replace( --bring back non-leading zeroes
+                ltrim( --trim leading spaces
+                replace( --replace 0 with space
+                substring(data,67,64) 
+                ,0,' ')
+                )
+                ,' ',0)
+            ,16,10)
+        AS bridged_token_amount_raw
+        , '0x' || substring(data,27,40) AS recipient_address
+        , '' AS trace_address
+        , l.index AS evt_index
+        , l.contract_address AS project_contract_address
+        , NULL AS transfer_id
+        , 10 AS source_chain_id
+        , 1 AS destination_chain_id
+        ,DENSE_RANK() OVER (PARTITION BY tx_hash ORDER BY index ASC) AS tf_index
+        
+        FROM {{ source ('optimism', 'logs') }} l
+        WHERE topic1 = '0x73d170910aba9e6d50b102db522b1dbcd796216f5128b445aa2135272886497e' --Withdrawal Initiated
+        -- Other potentially helpful filters
+        AND (topic2 IS NOT NULL) AND (topic3 IS NOT NULL) AND (topic4 IS NOT NULL) AND (data IS NOT NULL)
+        {% if is_incremental() %}
+        AND block_time > NOW() - interval '14 days'
+        {% endif %}
+        
+        ) a
+    LEFT JOIN {{ ref ('ovm_optimism_cross_domain_messenger_messages') }} m
+        ON a.tf_type = m.msg_type
+        AND a.block_time = m.l2_block_time
+        AND a.block_number = m.l2_block_number
+        AND a.tx_hash = m.l2_tx_hash
+        AND a.tf_index = m.msg_index
+        {% if is_incremental() %}
+        AND m.l2_block_time > NOW() - interval '14 days'
+        {% endif %}
+)
+
+SELECT
+ 'optimism' as chain_data_source
+, 'Standard Bridge' AS project
+, tf.block_date
+, tf.block_time
+, source_chain_id
+, destination_chain_id
+, cid_source.chain_name AS source_chain_name
+, cid_dest.chain_name AS destination_chain_name
+, erc.symbol AS bridged_token_symbol
+, bridged_token_amount_raw / POWER(10,erc.decimals) AS bridged_token_amount
+, 0 AS bridged_token_fee_amount
+, p.price*( bridged_token_amount_raw / POWER(10,erc.decimals) ) AS bridged_amount_usd
```

Comment:
We should make sure to ensure types by explicitly casting amount using CAST(`x` AS DOUBLE) for example

Reply:
on this you mean the `...token_amount_raw`? Or decimals as well?